Fixing tests

ahcorde opened the pull request.

ahcorde committed:
Fixing issues with Logger variable name

ahcorde committed:
fixing utils and robot_model tests

vmayoral commented:
Let's wait for the CI

vmayoral reviewed (changes requested):
On a second look , this is actually not acceptable. You've changed the LOGGER names and that was agreed through a lot of iterations. Can you please revert this @ahcorde?

I mean, no changes to `LOGGER` variable names should be done.

ahcorde commented:
I can't otherwise it's not compiling

vmayoral commented on moveit_core/background_processing/src/background_processing.cpp:
```diff
@@ -80,13 +80,13 @@ void BackgroundProcessing::processingThread()
       try
       {
-        RCLCPP_DEBUG(LOGGER, "Begin executing '%s'", action_name.c_str());
+        RCLCPP_DEBUG(LOGGER_BACKGROUND_PROCESSING, "Begin executing '%s'", action_name.c_str());
```
This shouldn't be reverted.

vmayoral commented on moveit_core/background_processing/src/background_processing.cpp:
```diff
@@ -42,7 +42,7 @@ namespace moveit
 {
 // Logger
-rclcpp::Logger LOGGER = rclcpp::get_logger("background_processing");
+rclcpp::Logger LOGGER_BACKGROUND_PROCESSING = rclcpp::get_logger("background_processing");
```
This shouldn't be changed either.

vmayoral commented on moveit_core/background_processing/src/background_processing.cpp:
```diff
@@ -80,13 +80,13 @@ void BackgroundProcessing::processingThread()
       catch (std::exception& ex)
       {
-        RCLCPP_ERROR(LOGGER, "Exception caught while processing action '%s': %s", action_name.c_str(), ex.what());
+        RCLCPP_ERROR(LOGGER_BACKGROUND_PROCESSING, "Exception caught while processing action '%s': %s", action_name.c_str(), ex.what());
```
etc.

vmayoral commented:
> I can't otherwise it's not compiling

Let's look at this together.

LanderU commented:
```bash
libmoveit_robot_model.a(robot_model.cpp.o): In function `moveit::core::RobotModel::constructShape(urdf::Geometry const*)':
robot_model.cpp:(.text+0x9116): undefined reference to `shapes::Sphere::Sphere(double)'
robot_model.cpp:(.text+0x918d): undefined reference to `shapes::Box::Box(double, double, double)'
robot_model.cpp:(.text+0x91d8): undefined reference to `shapes::Cylinder::Cylinder(double, double)'
robot_model.cpp:(.text+0x925d): undefined reference to `shapes::createMeshFromResource(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, Eigen::Matrix<double, 3, 1, 0, 3, 1> const&)'
libmoveit_robot_model.a(link_model.cpp.o): In function `moveit::core::LinkModel::setGeometry(std::vector<std::shared_ptr<shapes::Shape const>, std::allocator<std::shared_ptr<shapes::Shape const> > > const&, std::vector<Eigen::Transform<double, 3, 1, 0>, Eigen::aligned_allocator<Eigen::Transform<double, 3, 1, 0> > > const&)':
link_model.cpp:(.text+0x720): undefined reference to `shapes::computeShapeExtents(shapes::Shape const*)'
link_model.cpp:(.text+0x77e): undefined reference to `typeinfo for shapes::Mesh'
link_model.cpp:(.text+0x785): undefined reference to `typeinfo for shapes::Shape'
collect2: error: ld returned 1 exit status
make[2]: *** [robot_model/test_moveit_robot_model] Error 1
```

LanderU commented:
OS X:

```bash
CMake Error in robot_model/CMakeLists.txt:
  Imported target "resource_retriever::resource_retriever" includes
  non-existent path
    "/Users/osrf/jenkins-agent/workspace/packaging_osx/ws/install/include"
  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:
  * The path was deleted, renamed, or moved to another location.
  * An install or uninstall procedure did not complete successfully.
  * The installation package was faulty and references files it does not
  provide.
```

ahcorde committed:
Fixing robot_state tests

ahcorde committed:
Fix trajectory_processing tests

ahcorde committed:
Fixed collision detection tests

ahcorde committed:
Fixing collision detection fcl tests

ahcorde committed:
Converting all libraries to SHARED

ahcorde committed:
Fixing kinematic constraints tests

ahcorde committed:
Fixing constraint samplers tests

ahcorde committed:
Fixing planning scene tests

ahcorde committed:
Updating order in compilation

ahcorde committed:
Fixing constraint_samplers tests

ahcorde committed:
Fixing trajectory_processing tests

ahcorde committed:
Fixing distance_field tests

ahcorde committed:
Fixing collision distance field tests

ahcorde committed:
adding libs to planning_scene

vmayoral commented:
@ahcorde, this is till failing. What's the status?

ahcorde committed:
Fixed logger in move_group

ahcorde committed:
adding node as argument lookupParam()

ahcorde committed:
adding compiler options

ahcorde committed:
fixing logger variables

ahcorde committed:
updating kinematics_plugin_loader

ahcorde committed:
updating rdf_loader

ahcorde committed:
updating robot_model_loader - adding node to constructors

ahcorde committed:
updating planning_scene_monitor - adding node to constructors

ahcorde committed:
porting moveit_kinematics

anasarrak commented on moveit_ros/planning/rdf_loader/src/rdf_loader.cpp:
```diff
@@ -53,23 +54,31 @@
+  }, qos);
+
+  while(content.length()<1){
+    printf("Waiting for Robot model topic! Did you remap '%s'?\n", robot_description.c_str());
```
Please lets remove the printf and use RCLCPP_WARN or RCLCPP_INFO.

anasarrak commented on moveit_ros/planning/rdf_loader/src/rdf_loader.cpp:
```diff
@@ -83,11 +92,16 @@ rdf_loader::RDFLoader::RDFLoader(const std::string& robot_description)
+  }, qos);
+
+  while(scontent.length()<1){
+    printf("Waiting for Robot model semantic topic! Did you remap '%s'?\n", std::string(robot_description + "_semantic").c_str());
```
Same as https://github.com/AcutronicRobotics/moveit2/pull/56/commits/2cf435bf4dc98876a1253d3d5c53a28ebf8f15dd#r280375688

anasarrak commented on moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp:
```diff
@@ -47,11 +47,25 @@ RobotModelLoader::RobotModelLoader(const std::string& robot_description, bool lo
 
 RobotModelLoader::RobotModelLoader(const Options& opt)
 {
+  printf("RobotModelLoader Node is not declare!!\n");
```
Same as https://github.com/AcutronicRobotics/moveit2/pull/56/commits/2cf435bf4dc98876a1253d3d5c53a28ebf8f15dd#r280375688

anasarrak commented on moveit_ros/planning/robot_model_loader/src/robot_model_loader.cpp:
```diff
@@ -47,11 +47,25 @@ RobotModelLoader::RobotModelLoader(const std::string& robot_description, bool lo
 {
   Options opt(robot_description);
   opt.load_kinematics_solvers_ = load_kinematics_solvers;
+  printf("RobotModelLoader Node is not declare!!\n");
```
same https://github.com/AcutronicRobotics/moveit2/pull/56/commits/2cf435bf4dc98876a1253d3d5c53a28ebf8f15dd#r280375688

anasarrak commented on moveit_core/robot_state/src/robot_state.cpp:
```diff
@@ -1051,7 +1051,7 @@ void RobotState::getRobotMarkers(visualization_msgs::msg::MarkerArray& arr, cons
   // ros::Time tm = ros::Time::now();
-
+  rclcpp::Clock clock_ros;
   builtin_interfaces::msg::Time tm = clock_ros.now();
```
```suggestion
   builtin_interfaces::msg::Time tm = rclcpp::Clock().now();
```
What about this change?

ahcorde committed:
adding node and fixing logger in kinematics_base

ahcorde committed:
Fix Cmakelists BUILD_TESTING

ahcorde committed:
Adding kdl_kinematics_plugin

ahcorde committed:
Converting time to seconds

ahcorde committed:
Comment lookupParam method

ahcorde committed:
resource retriever from sources

Update osx-compile-moveit2.sh

Update osx-install-ros2-binary.sh

Update moveit2.repos

Update package.xml

Update osx-compile-moveit2.sh

LanderU committed:
Restore moveit2.repos and remove it from OSX

ahcorde committed:
Update package.xml

ahcorde committed:
Update package.xml

ahcorde committed:
Adding moveit_core/collision_detection dependency

ahcorde commented:
CI is configure with Crystal patch4, but we are using dashing (beta). We need to fix CI

ahcorde committed:
kinematic_constraints adding library dependencies

ahcorde committed:
planning_scene_monitor adding library dependencies

ahcorde committed:
robot_model_loader adding library dependencies

ahcorde committed:
planning_interface adding library dependencies

ahcorde committed:
adding trajectory_execution_manager available kinematics plugin

ahcorde committed:
move_group_interface adding more methods and compiling demo

ahcorde committed:
Adding node as argument common_planning_interface_objects

vmayoral commented:
@ahcorde, shall we close this one?

ahcorde commented:
sure
